### PR TITLE
Users/visuri/failurecriteris

### PR DESCRIPTION
**Pull request by vinitsuri**
-Added failure criteria of Avg.Response Time(ms)

### CODE REVIEW

File: Tasks/QuickPerfTest/CltThresholdValidationHelper.ps1 [hunk 1]
```diff
@@ -0,0 +1,113 @@
+Class ThresholdRule
```
Comment (cltshivash) [inline Tasks/QuickPerfTest/CltThresholdValidationHelper.ps1]: >Class [](start = 0, length = 5)

check the version support ..? base support should be for 2012 R2

File: Tasks/QuickPerfTest/CltThresholdValidationHelper.ps1 [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+function ValidateThreshold($cltAccountUrl,$headers,[ThresholdRule]$thresholdRule,$testRunId,$counterInstances)
+{
+ $counterInstanceId = GetCounterInstanceId $counterInstances $thresholdRule.counterInstanceName
+ $samplesRemaining = $true
+ $startInterval = 0
+ $endInterval = 2000
+ $violationCount = 0
+ While($samplesRemaining -eq $true)
```
Comment (cltshivash) [inline Tasks/QuickPerfTest/CltThresholdValidationHelper.ps1]: >While($samplesRemaining -eq $true) [](start = 1, length = 34)

close with Prachi. the check is stringent here.. (at a interval level)..

**Comment by cltshivash** [comment]
:clock1:

File: Tasks/QuickPerfTest/CltThresholdValidationHelper.ps1 [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+function ValidateThresholdRule($cltAccountUrl,$headers,[ThresholdRule]$thresholdRule,$testRunId)
+{
+ $counterInstances = Get-CounterInstances $cltAccountUrl $headers $testRunId
+ return ValidateThreshold $cltAccountUrl $headers $thresholdRule $testRunId $counterInstances
+}
+
+function ValidateThreshold($cltAccountUrl,$headers,[ThresholdRule]$thresholdRule,$testRunId,$counterInstances)
+{
+ $counterInstanceId = GetCounterInstanceId $counterInstances $thresholdRule.counterInstanceName
```
Comment (dpksinghal) [inline Tasks/QuickPerfTest/CltThresholdValidationHelper.ps1]: >GetCounterInstanceId [](start = 22, length = 20)

Minor: lets keep one pattern in method names either '<verb>-<resource>' or '<Verb><Resource>'

File: Tasks/QuickPerfTest/CltThresholdValidationHelper.ps1 [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+
+function GetCounterInstanceId($counterInstances,$counterName)
+{
+ ForEach($counterInstance in $counterInstances.value)
+ {
+  if($counterInstance.counterName -eq $thresholdRule.counterInstanceName)
+  {
+   $counterInstanceId = $counterInstance.counterInstanceId
```
Comment (dpksinghal) [inline Tasks/QuickPerfTest/CltThresholdValidationHelper.ps1]: >$c [](start = 3, length = 2)

shouldnt you break the loop here?

File: Tasks/QuickPerfTest/CltThresholdValidationHelper.ps1 [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  }
+ }
+ return $violationCount
+}
+
+function GetViolationsCount($samples, $thresholdValue, $isHigherBetter)
+{
+ if($isHigherBetter -eq $false)
+ {
+ $thresholdviolatedSamples = $samples.values.values.computedValue -gt $thresholdValue
```
Comment (dpksinghal) [inline Tasks/QuickPerfTest/CltThresholdValidationHelper.ps1]: >$thresholdviolatedSamples [](start = 1, length = 25)

will this be not just a Boolean?

File: Tasks/QuickPerfTest/CltThresholdValidationHelper.ps1 [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+ $violationCount = 0
+ While($samplesRemaining -eq $true)
+ {
+  $samples = Get-Samples $cltAccountUrl $headers $testRunId $counterInstanceId $startInterval $endInterval
+  if($samples -and $samples.count -gt 0)
+  {
+   $newViolations =  GetViolationsCount $samples $thresholdRule.thresholdValue $thresholdRule.isHigherBetter
+   $violationCount = $violationCount + $newViolations
+  }
+  if($samples.count -ne 2000)
```
Comment (dpksinghal) [inline Tasks/QuickPerfTest/CltThresholdValidationHelper.ps1]: >($samples.count -ne 2000) [](start = 4, length = 25)

What is the strategy here to come out of the loop? Where are you accumulating the count of samples after each iteration of the while loop? What is this magic number 2000 to break the loop?

**Pull request closed without merging**